### DISCUSSION TITLE
Help with file uploads

### DISCUSSION THREAD

**Discussion by mpringlehlk** [johannschopplich/nuxt-api-party#19] [Q&A] [answered]
Hi,

We are using `$apiParty` to send data to our secure API, but I've run into an issue when trying to upload files. It appears that the File data is being sent as a generic blob and the filename is being lost. 

We are building the body in the usual way, creating a variable to hold a `new FormData()` instance, then appending the file to it.

If I use a standard Nuxt client-side `$fetch` call with the exact same formData, the file content is sent properly and everything works as expected.

When comparing the POST requests in the browser, I noticed the $apiParty body is being sent as JSON, so I tried manually setting a "Content-Type": "multipart/form-data" on the request, but then I get an error back from our API stating the form data boundaries are missing.

Any advice on how to resolve this issue would be most welcome. Thanks!

**Comment by johannschopplich** [comment, marked as answer]
Hi there!

Glad to hear you're using Nuxt API Party in production. I recommend you to read the [How It Works](https://nuxt-api-party.byjohann.dev/guide/how-it-works.html) section again to get a better understanding about how the Nuxt server route to proxy your request works.

In a nutshell: The request has to be serialized for the `/api/__api_party` route. Inside that route, we then construct the `FormData` to send to your actual API. Thus, adding `multipart/form-data` as `Content-Type` won't have any effect, since a JSON object is expected by `/api/__api_party`.

Having said that, I have added support for `name` if the Blob is indeed of type `File`: https://github.com/johannschopplich/nuxt-api-party/commit/6d20ca4552d5ef14acbe234840c7f8799dd3203b#diff-16671b8e02b11955397ac6e8f00650f5dd3e959171b89942e20c9e6372f06776

Please test again with the latest version. Thanks!

**Reply by mpringlehlk** [reply]
That is fantastic, Johann! 

Just tested the latest version and all is well. Thank you very much for the quick fix. This proxy has been very useful to us.

-Mike
